File: BaekJoon Online Judge/C++/Step06-Function/2447.cpp
//
//  2447.cpp
//  Step6-Function
//
//  Created by 김예빈 on 2019. 8. 31..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

void printStar(int, int, int);

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, i, j;
    
    cin >> n;
    
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++)
            printStar(i, j, n);
        
        cout << '\n';
    }
    
    return 0;
}

void printStar(int x, int y, int size) {
    int mid = size / 3;
    
    if (((x / size) % 3 == 1) && ((y / size) % 3 == 1))
        cout << ' ';
    else {
        if (mid == 0)
            cout << '*';
        else
            printStar(x, y, mid);
    }
}

File: BaekJoon Online Judge/Swift/Step20-Recursion/2447.swift
//
//  2447.swift
//  Step10-Recursion
//
//  Created by Yebin Kim on 2019/11/24.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

// Timeout
func printStar(_ x: Int, _ y: Int, _ size: Int) {
    let mid = size / 3
    
    if ((x / size) % 3 == 1) && ((y / size) % 3 == 1) {
        print(" ", terminator: "")
    } else {
        if mid == 0 {
            print("*", terminator: "")
        } else {
            printStar(x, y, mid);
        }
    }
}

if let N = Int(readLine()!) {
    for i in 0..<N {
        for j in 0..<N {
            printStar(i, j, N)
        }
        
        print()
    }
}
